File: snobot_sim_java/java/snobot_sim_java/com/snobot/simulator/simulator_components/accelerometer/ThreeAxisAccelerometer.java
package com.snobot.simulator.simulator_components.accelerometer;

import com.snobot.simulator.SensorActuatorRegistry;

public class ThreeAxisAccelerometer
{
    protected final AccelerometerWrapper mXWrapper;
    protected final AccelerometerWrapper mYWrapper;
    protected final AccelerometerWrapper mZWrapper;

    public ThreeAxisAccelerometer(int aBasePort, String aBaseName)
    {
        mXWrapper = new AccelerometerWrapper(aBaseName + "X");
        mYWrapper = new AccelerometerWrapper(aBaseName + "Y");
        mZWrapper = new AccelerometerWrapper(aBaseName + "Z");

        SensorActuatorRegistry.get().register(mXWrapper, aBasePort + 0);
        SensorActuatorRegistry.get().register(mYWrapper, aBasePort + 1);
        SensorActuatorRegistry.get().register(mZWrapper, aBasePort + 2);
    }

    public double getX()
    {
        return mXWrapper.getAcceleration();
    }

    public double getY()
    {
        return mYWrapper.getAcceleration();
    }

    public double getZ()
    {
        return mZWrapper.getAcceleration();
    }

}

File: snobot_sim/src/main/native/cpp/SnobotSim/SimulatorComponents/Accelerometer/ThreeAxisAccelerometer.cpp
/*
 * ThreeAxisAccelerometer.cpp
 *
 *  Created on: Jul 12, 2017
 *      Author: preiniger
 */

#include "SnobotSim/SimulatorComponents/Accelerometer/ThreeAxisAccelerometer.h"
#include "SnobotSim/SensorActuatorRegistry.h"

ThreeAxisAccelerometer::ThreeAxisAccelerometer(int aPort, const std::string& aBaseName) :
    mXWrapper(new AccelerometerWrapper(aBaseName + " X")),
    mYWrapper(new AccelerometerWrapper(aBaseName + " Y")),
    mZWrapper(new AccelerometerWrapper(aBaseName + " Z"))
{
    SensorActuatorRegistry::Get().Register(aPort + 0, mXWrapper);
    SensorActuatorRegistry::Get().Register(aPort + 1, mYWrapper);
    SensorActuatorRegistry::Get().Register(aPort + 2, mZWrapper);
}

ThreeAxisAccelerometer::~ThreeAxisAccelerometer()
{

}


double ThreeAxisAccelerometer::GetX()
{
    return mXWrapper->GetAcceleration();
}

double ThreeAxisAccelerometer::GetY()
{
    return mYWrapper->GetAcceleration();
}

double ThreeAxisAccelerometer::GetZ()
{
    return mZWrapper->GetAcceleration();
}

std::shared_ptr<AccelerometerWrapper>& ThreeAxisAccelerometer::GetXWrapper()
{
    return mXWrapper;
}

std::shared_ptr<AccelerometerWrapper>& ThreeAxisAccelerometer::GetYWrapper()
{
    return mYWrapper;
}

std::shared_ptr<AccelerometerWrapper>& ThreeAxisAccelerometer::GetZWrapper()
{
    return mZWrapper;
}

const std::shared_ptr<AccelerometerWrapper>& ThreeAxisAccelerometer::GetXWrapper() const
{
    return mXWrapper;
}

const std::shared_ptr<AccelerometerWrapper>& ThreeAxisAccelerometer::GetYWrapper() const
{
    return mYWrapper;
}

const std::shared_ptr<AccelerometerWrapper>& ThreeAxisAccelerometer::GetZWrapper() const
{
    return mZWrapper;
}
